File: src/main/java/es/alejandro12120/kqueue/types/Queue.java
package es.alejandro12120.kqueue.types;

import org.bukkit.entity.Player;

public class Queue {

    private SList list;

    public Queue() {
        this.list = new SList();
    }

    public void enqueue(Player element) {
        this.list.addLast(element);
    }

    public Player dequeue() {
        return this.list.removeFirst();
    }

    public Player front() {
        return this.list.getHead().getElement();
    }

    public boolean isEmpty() {
        return this.list.isEmpty();
    }

    public int size() {
        return this.list.getSize();
    }

}

File: src/main/java/es/alejandro12120/kqueue/types/Queue.kt
package es.alejandro12120.kqueue.types

import org.bukkit.entity.Player

data class Queue(

    var list: SList

) {

    constructor() : this(SList())

    fun enqueue(element: Player) {
        this.list.addLast(element)
    }

    fun dequeue(): Player? {
        return this.list.removeFirst()
    }

    fun front(): Player? {
        return this.list.head?.element
    }

    fun getPosition(player: Player): Int {
        return this.list.getPosition(player.name)
    }

    fun getPlayers(): List<Player> {
        return this.list.getPlayers()
    }

    fun isEmpty(): Boolean {
        return this.list.isEmpty()
    }

    fun size(): Int {
        return this.list.size
    }
}